typegate/tests/utils/tg_deploy_script.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+import { BasicAuth, tgDeploy } from "@typegraph/sdk/tg_deploy.js";
+import * as path from "std/path/mod.ts";
+
+const cwd = Deno.args[0];
+const PORT = Deno.args[1];
+const modulePathStr = Deno.args[2];
+const secretsStr = Deno.args[3];
+
+const gate = `http://localhost:${PORT}`;
+const auth = new BasicAuth("admin", "password");
+
+// resolve the module
+const moduleName = path.basename(modulePathStr);
+const tgPath = path.join(cwd, moduleName);
+
+const module = await import(tgPath);
+if (!module.tg) {
+  throw new Error(`No typegraph found`);
+}
+
+const secrets = JSON.parse(secretsStr);
+
+const tg = module.tg;
+const { serialized, typegate: _gateResponseAdd } = await tgDeploy(tg, {
+  baseUrl: gate,
+  auth,
+  artifactsConfig: {
+    prismaMigration: {
+      globalAction: {
+        create: true,
+        reset: false,
+      },
+      migrationDir: "prisma-migrations",
+    },
+    dir: cwd,
+  },
+  typegraphPath: tgPath,
+  secrets: secrets,
```

Comment:
Btw I think `globalAction`, `codegen` and `disableArtifactResolution` should all be configurable from the envs.
This might come handy in some context.
Enabling a custom setting for the entire artifact resolution config from env is another option though!

Reply:
oww, that's a good point! I guess we can get the value stored in env, if it's there, if not we can use a default value which is already being used now. wdyt?